transform-plugins/src/main/java/io/cdap/plugin/XMLToJSON.java

```diff
@@ -57,10 +62,38 @@ public XMLToJSON(Config config) {
     this.config = config;
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    generateOutputSchema();
+    if (context.getInputSchema() == null || context.getInputSchema().getFields() == null ||
+      context.getOutputSchema() == null || context.getOutputSchema().getFields() == null) {
+      return;
+    }
+
+    // Output operation list includes:
+    // - config.inputField mapping to all output fields (one-to-many)
+    // - other fields in inputSchema that map to themselves (1-to-1)
+    List<String> identityFields = context.getInputSchema().getFields().stream().map(Schema.Field::getName)
+      .filter(field -> outputSchema.getField(field) != null && !field.equals(config.inputField))
+      .collect(Collectors.toList());
+
+    List<FieldOperation> outputList = new ArrayList<>(
+      TransformLineageRecorderUtils.generateOneToMany(config.inputField,
+        TransformLineageRecorderUtils.getFields(context.getOutputSchema()),
```

Comment:
This is not true, config.inputField only maps to all the non-identity fields in the output schema